frontend/app/src/utils/history.ts

```diff
@@ -0,0 +1,144 @@
+import makeBlockie from 'ethereum-blockies-base64';
+import i18n from '@/i18n';
+import {
+  ActionDataEntry,
+  transactionEventProtocolData,
+  transactionEventTypeData
+} from '@/store/history/consts';
+import { EthTransactionEventEntry } from '@/store/history/types';
+import {
+  HistoryEventSubType,
+  HistoryEventType,
+  TransactionEventProtocol,
+  TransactionEventType
+} from '@/types/transaction';
+
+export const getEventType = (
+  event: EthTransactionEventEntry
+): TransactionEventType | undefined => {
+  const { eventType, eventSubtype } = event;
+
+  // TransactionEventType.GAS
+  if (
+    eventType === HistoryEventType.SPEND &&
```

Comment:
Wouldn't it be better if we used a map of event types instead: `{ [type]: {[subtype]: TransactionEventType}| TransactionEventType}`  and check the map of fall to default?

As we add more and more types/subtypes here the if will become hard to manage